modules/set.js

```diff
@@ -0,0 +1,37 @@
+import isObject from './isObject.js';
+import toPath from './_toPath.js';
+import contains from './contains.js';
+
+
+var arrayIndex = /^\d+$/;
+
+// Internal function of `set`.
+function deepSet(obj, path, value) {
+  var key = path[0];
+
```

Comment:
I think this is the right place to check that `key` is not one of `'__proto__'`, `'constructor'` or `'prototype'`.

Reply:
I'm afraid it will mutate before throwing an error.
```javascript
const obj = { value: 50 };

try {
  _.set(obj, [ 'key', '__proto__' ], {});
} catch(e) {
  console.log(e.message);  //  Prototype assignment attempted
  console.log(obj);        //  { value: 50, key: {} }  -  `obj` was mutated :(
}
```

Reply:
Yeah, I realized that as well. I can think of three options (from least to most preferred, in my opinion):

1. Stick with `_.contains`, accept the double iteration and the greater weight after treeshaking.
2. Accept that `obj` gets somewhat corrupted. At least the prototype pollution attempt is detected and unsuccessful.
3. Switch to postorder assignment as I suggested before, which has neither of these problems.

I didn't mention it because I was ready to accept option 2, but I would prefer option 3.